csharp/private/wrappers/dotnet.cc

```diff
@@ -0,0 +1,97 @@
+#include <iostream>
+#include <sstream>
+#include <string>
+
+#ifdef _WIN32
+#include <errno.h>
+#include <process.h>
+#include <windows.h>
+#else  // not _WIN32
+#include <stdlib.h>
+#include <unistd.h>
+#endif  // _WIN32
+
+#include "tools/cpp/runfiles/runfiles.h"
+
+using bazel::tools::cpp::runfiles::Runfiles;
+
+std::string evprintf(std::string name, std::string path) {
+  std::stringstream ss;
+  ss << name << "=" << path;
+  return ss.str();
+}
+
+int main(int argc, char** argv) {
+  std::string error;
+
+  auto runfiles = Runfiles::Create(argv[0], &error);
+
+  if (runfiles == nullptr) {
+    std::cerr << "Couldn't load runfiles: " << error << std::endl;
+    return 101;
+  }
+
+  auto dotnet = runfiles->Rlocation("{DotnetExe}");
+  if (dotnet.empty()) {
+    std::cerr << "Couldn't find the .NET runtime" << std::endl;
+    return 404;
+  }
+
+  // Get the name of the directory containing dotnet.exe
+  auto dotnetDir = dotnet.substr(0, dotnet.find_last_of("/\\"));
+
+  /*
+  dotnet and nuget require these environment variables to be set
+  without them we cannot build/run anything with dotnet.
+
+  dotnet: HOME, DOTNET_CLI_HOME, APPDATA, PROGRAMFILES
+  nuget: TMP, TEMP, USERPROFILE
+  */
+  std::vector<std::string> envvars;
+  envvars.push_back(evprintf("HOME", dotnetDir));
```

Comment:
Use list initialization:

```
std::vector<std::string> envvars = {
  "foo",
  "bar",
  ...
}
```

Reply:
For commit 80e56d3d468ac767c98b22231df22891057d25aa I tried switching over to list initialization, but MacOS didn't love that. I suspect I might need to cast every single list item into a `const_char` to get it to work.

Reply:
Looks like that was it